azkaban-common/src/main/java/azkaban/AzkabanCommonModule.java

```diff
@@ -82,6 +83,7 @@ protected void configure() {
     bind(DataSource.class).to(AzkabanDataSource.class);
     bind(ExecutorLoader.class).to(JdbcExecutorLoader.class);
     bind(MetricRegistry.class).in(Scopes.SINGLETON);
+    bind(ExecutorApiClient.class).in(Scopes.SINGLETON);
```

Comment:
Please use Singleton annotation and remove this explicit binding.
We plan to remove existing such usage pattern.

Reply:
You might want to refer to #1428 .